src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperUsageDescriptorFactory.cs

```diff
@@ -0,0 +1,155 @@
+            // Couldn't resolve a valid assemblyLocation.
+            if (string.IsNullOrEmpty(assemblyLocation))
+            {
+                return null;
+            }
+
+            var xmlDocumentationFile = GetXmlDocumentationFile(assembly, assemblyLocation);
+
+            // Only want to process the file if it exists.
+            if (xmlDocumentationFile != null)
+            {
+                var documentationProvider = new XmlDocumentationProvider(xmlDocumentationFile.FullName);
+
+                var summary = documentationProvider.GetSummary(id);
+                var remarks = documentationProvider.GetRemarks(id);
+
+                if (!string.IsNullOrEmpty(summary) || !string.IsNullOrEmpty(remarks))
+                {
+                    return new TagHelperUsageDescriptor(summary, remarks);
+                }
+            }
+
+            return null;
+        }
+
+        private static FileInfo GetXmlDocumentationFile(Assembly assembly, string assemblyLocation)
+        {
+            try
+            {
+                var assemblyDirectory = Path.GetDirectoryName(assemblyLocation);
+                var assemblyName = Path.GetFileName(assemblyLocation);
+                var assemblyXmlDocumentationName = Path.ChangeExtension(assemblyName, ".xml");
+
+                // Check for a localized XML file for the current culture.
+                var xmlDocumentationFile = GetLocalizedXmlDocumentationFile(
+                    CultureInfo.CurrentCulture,
+                    assemblyDirectory,
+                    assemblyXmlDocumentationName);
+
+                if (xmlDocumentationFile == null)
+                {
+                    // Check for a culture-neutral XML file next to the assembly
+                    xmlDocumentationFile = new FileInfo(
+                        Path.Combine(assemblyDirectory, assemblyXmlDocumentationName));
+
+                    if (!xmlDocumentationFile.Exists)
+                    {
+                        xmlDocumentationFile = null;
+                    }
+                }
+
+                return xmlDocumentationFile;
+            }
+            catch (ArgumentException)
+            {
+                // Could not resolve XML file.
+                return null;
+            }
+        }
+
+        private static IEnumerable<string> ExpandPaths(
+            CultureInfo culture,
+            string assemblyDirectory,
+            string assemblyXmlDocumentationName)
+        {
+            do
+            {
+                var cultureName = culture.Name;
+                var cultureSpecificFileName =
+                    Path.ChangeExtension(assemblyXmlDocumentationName, cultureName + ".xml");
+
+                yield return Path.Combine(assemblyDirectory, cultureSpecificFileName);
+                yield return Path.Combine(assemblyDirectory, cultureName, assemblyXmlDocumentationName);
+                yield return Path.Combine(assemblyDirectory, cultureName, cultureSpecificFileName);
```

Comment:
I left comments out of my sample code. Please restore the link to `https://msdn.microsoft.com/en-us/library/sb6a8618.aspx#cpconpackagingdeployingresourcesanchor1` (above the `do`) and the short descriptions of each path e.g. `// A culture-specific XML file next to the assembly.` (with blank lines above each comment).